middleman_protocol/middleman_protocol/constants.py

```diff
@@ -0,0 +1,54 @@
+from enum import IntEnum
+from enum import unique
+
+
+@unique
+class PayloadType(IntEnum):
+    GOLEM_MESSAGE               = 0  # A serialized Golem message.
+    ERROR                       = 1  # An error code and an error message.
+    AUTHENTICATION_CHALLENGE    = 2  # A random string of bytes.
+    AUTHENTICATION_RESPONSE     = 3  # A digital signature of the content sent as authentication challenge.
+
+
+@unique
+class ErrorCode(IntEnum):
+    InvalidFrame            = 0
+    InvalidFrameSignature   = 1
+    DuplicateFrame          = 2
+    InvalidPayload          = 3
+    UnexpectedMessage       = 4
+    AuthenticationFailure   = 5
+    ConnectionLimitExceeded = 6
+    MessageLost             = 7
+    ConnectionTimeout       = 8
+
+
+# Random sequence of bytes used as separator between messages. Should be placed after frame.
+FRAME_SEPARATOR = b'\x1d\xb7{\xb0\xb9\x16\xc8f/\xd7\xf0\xc7\x06\x07\x1e\xa2'
+
+FRAME_SEPARATOR_LENGTH = len(FRAME_SEPARATOR)
+
+FRAME_SIGNATURE_BYTES_LENGTH = 65
+FRAME_REQUEST_ID_BYTES_LENGTH = 4
+FRAME_PAYLOAD_LENGTH_BYTES_LENGTH = 4
+
+FRAME_PAYLOAD_LENGTH_STARTING_BYTE = (
+    FRAME_SIGNATURE_BYTES_LENGTH +
+    FRAME_REQUEST_ID_BYTES_LENGTH +
+    1  # 1 byte for payload_type
+)
+
+FRAME_PAYLOAD_STARTING_BYTE = FRAME_PAYLOAD_LENGTH_STARTING_BYTE + FRAME_PAYLOAD_LENGTH_BYTES_LENGTH
+
+ESCAPE_CHARACTER = b'\xa2\x1e\x07\xc7\xa2\x1e\x07\xc7\xa2\x1e\x07\xc7\xa2\x1e\x07\xc7'
+ESCAPE_SEQUENCES = {
+    ESCAPE_CHARACTER: ESCAPE_CHARACTER + b'\x00',
+    FRAME_SEPARATOR: ESCAPE_CHARACTER + b'\x01',
+}
+
+assert ESCAPE_CHARACTER != FRAME_SEPARATOR
+assert len(set(ESCAPE_SEQUENCES.values())) == len(ESCAPE_SEQUENCES)
+assert len(ESCAPE_CHARACTER) == len(FRAME_SEPARATOR)
```

Comment:
Either both are compared to one, or one of them is compared to one and to each other. So it is just one of two ways. If you insist I can change it, but I prefer it this way because of only one hardcoded value.

Reply:
I'd change it because even if the end result is the same,  it gives an impression that they must have the same length in general which is not true. One of the main purposes of assertions is documenting the assumptions after all.

I'd prefer to have two separate assertions, each checking that the respective constant has length of 1.